### ISSUE TITLE
[telemetry] HTTP session and servlet context lifecycle metrics in WebAppListener

### ISSUE THREAD

**Issue by phax** [phax/ph-oton#45]
## Goal

Emit HTTP session and servlet-context lifecycle metrics from `WebAppListener` in `ph-oton-core`.

Active session count is the most basic "is my application under load" signal, and ph-oton is already sitting on the two callbacks needed to produce it.

## Where

`ph-oton-core/src/main/java/com/helger/photon/core/servlet/WebAppListener.java`

The class implements both `ServletContextListener` and `HttpSessionListener`, and all four hooks already exist:

```java
public final void contextInitialized (@NonNull final ServletContextEvent aSCE)
public final void contextDestroyed (@NonNull final ServletContextEvent aSCE)
public final void sessionCreated (@NonNull final HttpSessionEvent aSessionEvent)
public final void sessionDestroyed (@NonNull final HttpSessionEvent aSessionEvent)
```

`contextInitialized` and `contextDestroyed` each already create a `StopWatch` and log the elapsed milliseconds at the end — the value is measured today and then thrown away.

## What to emit

Extend `com.helger.photon.core.CCoreTelemetry` (create it here if the UI-rendering issue has not landed yet):

| Constant | Value |
|---|---|
| `METRIC_SESSIONS_CREATED` | `photon.http.sessions.created` |
| `METRIC_SESSIONS_ACTIVE` | `photon.http.sessions.active` |
| `METRIC_STARTUP_DURATION` | `photon.app.startup.duration` |
| `METRIC_SHUTDOWN_DURATION` | `photon.app.shutdown.duration` |
| `UNIT_SESSION` | `{session}` |
| `UNIT_MILLIS` | `ms` |

- **Counter** `photon.http.sessions.created` — incremented in `sessionCreated`, no attributes.
- **Up-down counter** `photon.http.sessions.active` — `+1` in `sessionCreated`, `-1` in `sessionDestroyed`. Both must use the *same* (empty) attribute set so the counter nets out to zero, exactly as `LongRunningJobMetrics.JOBS_RUNNING` documents.
- **Histogram** `photon.app.startup.duration` in ms — recorded in `contextInitialized` from the existing `StopWatch`.
- **Histogram** `photon.app.shutdown.duration` in ms — recorded in `contextDestroyed` from the existing `StopWatch`.

## Why an up-down counter and not a gauge here

Unlike `LoggedInUserManager` (which owns an authoritative `m_aLoggedInUsers` map and should use a gauge), `WebAppListener` has no container-independent way to ask "how many sessions are there right now". The two callbacks are the only source of truth, so a balanced up-down counter is the correct instrument.

Be aware that a container may not deliver `sessionDestroyed` for every session in every crash/restart scenario — the counter is a good operational signal, not an accounting record. Say so in the Javadoc of the constant.

## Cardinality

No attributes on any of these instruments. Do **not** add the session ID, the servlet context path, or the server name — session IDs are unbounded, and the other two are constant per deployment and belong on the OpenTelemetry `Resource`, which the deploying application configures.

## Startup/shutdown as spans?

Deliberately **not** part of this issue. `contextInitialized` runs before the application's OpenTelemetry SDK is necessarily installed, so a span started there would frequently be dropped. The histograms are recorded at the *end* of the method and do not have that problem.

## POM

`ph-oton-core` gets `ph-telemetry` transitively via `ph-oton-mgrs`, but declare it directly since it is used directly (no `<version>` — the BOM is imported in the root POM):

```xml
<dependency>
  <groupId>com.helger.telemetry</groupId>
  <artifactId>ph-telemetry</artifactId>
</dependency>
```

## Acceptance criteria

- [ ] Up-down counter uses identical attributes on both the `+1` and the `-1` side.
- [ ] Startup/shutdown histograms reuse the existing `StopWatch` values; no second measurement is introduced.
- [ ] No attributes carrying session IDs or deployment-constant values.
- [ ] `mvn install` passes; `README.md` news bullet added.

---

### Shared conventions for all ph-telemetry work in ph-oton

**Facades.** `com.helger.telemetry.Telemetry` for spans, `com.helger.telemetry.TelemetryMetrics` for instruments. If no `ITelemetryTracerSPI` / `ITelemetryMeterSPI` is registered by the deploying application, everything degrades to cheap no-ops — so emit unconditionally, never guard with an "is telemetry enabled" flag.

**No OpenTelemetry in ph-oton.** Only `com.helger.telemetry:ph-telemetry` is ever added. `ph-telemetry-otel` and the OpenTelemetry SDK are the deploying application's business. The version comes from the `ph-telemetry-parent-pom` BOM already imported in the root `pom.xml`, so the new `<dependency>` needs no `<version>`.

**Reference implementation.** `ph-oton-mgrs/src/main/java/com/helger/photon/mgrs/longrun/` — `CLongRunningJobTelemetry` (names), `LongRunningJobMetrics` (instrument holder), `LongRunningJobTelemetry` (package-private emitter). Follow that layout.

**Constants: one class per module** (decided; `CLongRunningJobTelemetry` stays where it is for backwards compatibility since it is public API since 10.4.0):

| Module | Constants class |
|---|---|
| ph-oton-io | `com.helger.photon.io.CIOTelemetry` |
| ph-oton-app | `com.helger.photon.app.CAppTelemetry` |
| ph-oton-audit | `com.helger.photon.audit.CAuditTelemetry` |
| ph-oton-ajax | `com.helger.photon.ajax.CAjaxTelemetry` |
| ph-oton-api | `com.helger.photon.api.CAPITelemetry` |
| ph-oton-security | `com.helger.photon.security.CSecurityTelemetry` |
| ph-oton-mgrs | `CLongRunningJobTelemetry` (existing) + `com.helger.photon.mgrs.CMgrsTelemetry` (new) |
| ph-oton-core | `com.helger.photon.core.CCoreTelemetry` |
| ph-oton-connect | `com.helger.photon.connect.CConnectTelemetry` |
| ph-oton-uicore | `com.helger.photon.uicore.CUICoreTelemetry` |
| ph-oton-datatables | `com.helger.photon.uictrls.datatables.CDataTablesTelemetry` |

Each is `@Immutable`, `final`, with a private constructor, and groups constants under `// === span names ===`, `// === metric instrument names ===`, `// === attribute keys ===`, `// === metric units ===` comments, exactly as `CLongRunningJobTelemetry` does. Every constant gets a Javadoc comment.

**Naming.** Spans and metrics `photon.<area>.<thing>`, attributes `photon.<area>.<key>`. Units follow the OpenTelemetry convention: `ms` for durations, `{request}` / `{login}` / `{job}` / `{session}` for counts.

**Cardinality is a hard rule.** Metric attributes must be bounded and small. Anything unbounded — user IDs, session IDs, concrete URL paths, execution IDs, IP addresses, exception messages — goes on **spans only**, never on a metric. This mirrors `ATTR_JOB_TYPE` (metric) vs `ATTR_JOB_EXECUTION_ID` (span-only) in `CLongRunningJobTelemetry`.

**Span/thread binding.** A span is bound to the thread that created it (see the warning block at the end of `LongRunningJobTelemetry.onJobEnd`). Prefer `Telemetry.withSpan*` so start and end are guaranteed to happen on the same thread; only fall back to a manual `startSpan` + try/finally when the span genuinely has to outlive the method.

**Existing `StatisticsManager` handlers stay.** Where a site already emits ph-commons statistics, telemetry is added *alongside* them — do not remove or change the statistics handlers. The long-term strategy for the two systems is tracked separately.

**Coding style.** phax styleguide: Hungarian notation, `final` parameters, JSpecify `@NonNull`/`@Nullable`, `_` prefix for private methods, private methods declared before their callers, space before `(` and `<`, inline string concatenation in log calls (no SLF4J `{}` placeholders), Apache 2.0 header.

**Definition of done.** `mvn install` passes; a bullet is added to the `v10.5.0 - work in progress` section of `README.md` under "News and noteworthy".

**Comment by phax** [comment]
Part of #55 — see that issue for the shared conventions, the module-level constants class table and the POM impact overview.

**Comment by phax** [comment]
Implemented for **10.5.1**.

### What was added

| File | Kind |
|---|---|
| `ph-oton-core/.../core/CCoreTelemetry.java` | extended - 4 new metric names, `UNIT_SESSION` |
| `ph-oton-core/.../core/servlet/WebAppMetrics.java` | new, public - counter, up-down counter, 2 histograms |
| `ph-oton-core/.../core/servlet/WebAppListener.java` | modified - 4 call sites |

`ph-oton-core` already declares `com.helger.telemetry:ph-telemetry` directly since #41, so no POM change was needed.

### Emission

* Counter `photon.http.sessions.created` in `sessionCreated`, no attributes.
* Up-down counter `photon.http.sessions.active`: `+1` in `sessionCreated`, `-1` in `sessionDestroyed`. Both use `add (long)`, i.e. the identical empty attribute set, so the counter nets out to zero - the same rule `LongRunningJobMetrics.JOBS_RUNNING` documents.
* Histogram `photon.app.startup.duration` in `contextInitialized`.
* Histogram `photon.app.shutdown.duration` in `contextDestroyed`.

### No second measurement

Both `contextInitialized` and `contextDestroyed` called `aSW.stopAndGetMillis ()` inline inside their log statement. The value is now extracted into a local (`nInitMillis` / `nDestroyMillis`) which feeds both the histogram and the unchanged log line, so the stopwatch is still stopped exactly once and the logged number and the recorded number are by construction the same.

### Cardinality

No attributes on any of the four instruments. No session ID, no servlet context path, no server name - the class Javadoc of `WebAppMetrics` records that the latter two belong on the OpenTelemetry `Resource` that the deploying application configures.

The Javadoc of `METRIC_SESSIONS_ACTIVE` carries the caveat from the issue verbatim: a container does not necessarily deliver a "session destroyed" event for every session in every crash or restart scenario, so this is a good operational signal but no accounting record.

### Startup/shutdown spans

Deliberately not added, as the issue decided: `contextInitialized` runs before the application's OpenTelemetry SDK is necessarily installed, so a span started there would frequently be dropped. Both histograms are recorded at the *end* of their method and do not have that problem.

### Verification

Full reactor `mvn install` over all 23 modules: BUILD SUCCESS, no test failures. No unit test was added - `WebAppListener` needs a real servlet container to exercise these four callbacks, and the issue's acceptance criteria did not ask for one. `README.md` news bullet added under `v10.5.1 - work in progress`.
